### PR TITLE
Change visuals

### PR DESCRIPTION
**Pull request by KristianNelson**
Add new function that uses ggplot instead of tmap. Includes new graphs for city selection project. Also includes minor changes that were done offline to the count_teleconnections function.

### CODE REVIEW

File: R/map_vis.R [hunk 1]
```diff
@@ -294,3 +294,408 @@ plot_watershed <- function(data_dir,
...
+  ) -> state_ws_extent
+
+  map.p <- rasterToPoints(cropcover_city_watersheds_agg)
+  df <- data.frame(map.p)
+  colnames(df) <- c("Longitude", "Latitude", "Value")
+  left_join(df, cropclasses, by = "Value") -> df2
+
+  flood_control_dams %>% st_as_sf() -> flood_control_sf
+  watersheds_city %>% st_as_sf() -> watershed_sf
+  flood_control_sf[watershed_sf, ] -> select_dams
```
Comment (swd-turner) [inline R/map_vis.R:451]: I get this error message here:
```
Error in st_geos_binop("intersects", x, y, sparse = sparse, prepared = prepared) : 
  st_crs(x) == st_crs(y) is not TRUE
```

File: R/map_vis.R [hunk 2]
```diff
@@ -294,3 +294,408 @@ plot_watershed <- function(data_dir,
...
+        stat_ecdf(pad = F) -> cdf_plot
+      cdf_plot +
+        geom_point(data = layer_data(cdf_plot) %>% filter(x == crop_point),
+                   aes(x, y), col = "red", size = 5) +
+        xlab("Crop Fraction") +
+        ylab("") +
+        theme_classic() -> plot6
+      cowplot::plot_grid(plot5,plot6) -> graph_plot2
+
+      cowplot::plot_grid(plot1,plot2,graph_plot1,graph_plot2)
```
Comment (swd-turner) [inline R/map_vis.R:547]: Suggest re-ordering: Other cities, Number of power plants, Cropland fraction, Crop water consumption (as % of yield)

File: R/map_vis.R [hunk 3]
```diff
@@ -294,3 +294,408 @@ plot_watershed <- function(data_dir,
...
+
+  subset(power_plants_city, `Power Plant Type` %in% c("Hydropower", "Thermal")) -> power_plant_list
+  if(nrow(power_plant_list) == 0){
+    if(nrow(flood_control_dams) == 0){
+
+      ggplot(state_shape_trans)+
+        geom_sf(fill = "white")+
+        geom_sf(data = watersheds_city_trans, fill = "dodgerblue3", colour = "darkgrey") +
+        geom_sf(data = city_point,fill = "hotpink", colour = "black", size = 7, alpha = 0.7, shape = 22) +
+        theme(line = element_blank(),
```
Comment (swd-turner) [inline R/map_vis.R:481]: `theme_void()`

File: R/map_vis.R [hunk 4]
```diff
@@ -294,3 +294,408 @@ plot_watershed <- function(data_dir,
...
+              axis.title.y = element_blank(),
+              panel.background = element_blank(),
+              plot.title = element_text(hjust = 0.5)) +
+        ggtitle(paste0(city)) -> plot1
+
+      ggplot(df2) +
+        geom_raster(aes(x= Longitude, y=Latitude, fill=color)) +
+        scale_fill_identity() +
+        geom_sf(data = watersheds_city_trans, fill = "white",colour = "black", alpha = 0) +
+        theme(line = element_blank(),
```
Comment (swd-turner) [inline R/map_vis.R:495]: `theme_void()`

File: R/map_vis.R (hunk 1 above)
Comment (swd-turner) [reply R/map_vis.R:435]: still getting this error... any ideas? Is this running clean for you if you clean your environment and attempt on a few different cities?

Comment (KristianNelson) [reply R/map_vis.R:435]: Hmm...I am not getting this error, but I can try to add something in that would fix this if I did get the error. I think it is just the a projection problem between the two sf objects.

**Pull request merged**